servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -508,6 +517,23 @@ public Completable closeAsyncGracefully() {
 
     @Override
     public List<Entry<ResolvedAddress, List<C>>> usedAddresses() {
+        // If we're already in the executor we can't submit a task and wait for it without deadlock but
+        // the access is thread safe anyway so just go for it.
+        if (sequentialExecutor.isCurrentThreadDraining()) {
+            return sequentialUsedAddresses();
+        }
+        CompletableFuture<List<Entry<ResolvedAddress, List<C>>>> future = new CompletableFuture<>();
+        sequentialExecutor.execute(() -> future.complete(sequentialUsedAddresses()));
+        try {
+            // This method is just for testing, so it's fine to do some awaiting.
+            return future.get(5, TimeUnit.SECONDS);
+        } catch (Exception ex) {
+            throw new AssertionError("Failed to get results", ex);
+        }
```

Comment:
If we wanted to, this pattern could be lifted into a method of the form `T SequentialExecutor.submit(Callable<T>)`.